### ISSUE TITLE
Possibility to create a stream with events that were processed/skipped by a table

### ISSUE THREAD

**Issue by iljaiwjew** [confluentinc/ksql#9218]
Hi, it would be nice to have the following feature:
 
 We created the following stream:
 
```
CREATE STREAM RESULTS (
    key VARCHAR,
    result BIGINT
  ) WITH (
    KAFKA_TOPIC='results',
    VALUE_FORMAT='JSON'
    PARTITIONS=1
  );
```

e.g. we pushed to it messages like this:

```
{ key: 'John', result: 20 },
{ key: 'John', result: 30 },
{ key: 'Sean', result: 50 },
```

then we created a table:

```
CREATE TABLE LAST_VALID_RESULT_BY_KEY
  AS SELECT key, LATEST_BY_OFFSET(result) AS last_valid_result,
  FROM results
  WHERE (results.result > 10)
  GROUP BY RESULTS.KEY
  EMIT CHANGES;
 ```
 
 then we want to have a stream which is the same as ```RESULTS``` but every it's event is produced only after ```LAST_VALID_RESULT_BY_KEY``` already processed/skipped this event from ```RESULT``` stream
 
 Also, will appreciate if someone make a tip how to implement it now if it's possible

**Comment by Gerrrr** [comment]
@iljaiwjew I think creating a stream from `LAST_VALID_RESULT_BY_KEY` should get you desired behavior:

```sql
ksql> CREATE STREAM s2 (key string key, last_valid_result bigint) WITH (kafka_topic='LAST_VALID_RESULT_BY_KEY', value_format='JSON');

 Message
----------------
 Stream created
----------------

ksql> select * from s2;
+-------------------------------------------------------------------------------+-------------------------------------------------------------------------------+
|KEY                                                                            |LAST_VALID_RESULT                                                              |
+-------------------------------------------------------------------------------+-------------------------------------------------------------------------------+
|John                                                                           |20                                                                             |
|John                                                                           |30                                                                             |
|Sean                                                                           |50                                                                             |
Query Completed
Query terminated
```

**Comment by Gerrrr** [comment]
Yeah, I meant the internal topic created by `LAST_VALID_RESULT_BY_KEY`. You can see its name in `DESCRIBE <> EXTENDED` output:

```sql
ksql> describe LAST_VALID_RESULT_BY_KEY EXTENDED;

...
Kafka topic          : LAST_VALID_RESULT_BY_KEY (partitions: 1, replication: 1)
...
```

**Comment by iljaiwjew** [comment]
@Gerrrr 
got it, thank you.

But I need a little bit different behaviour. Stream that is created on the top of changelog topic will include only events that were inserted to the table as new key/value pairs and also they will be transformed according to the aggregation but i need a stream which is the same as source one(```RESULTS``` in my examle) but every event in it is emitted only after according event in source stream was handled by the table(```LAST_VALID_RESULT_BY_KEY```). In other words, there should be a mechanism to block events emitting until they are not handled by some other query

**Comment by Gerrrr** [comment]
I don't think ksqlDB has a convenient mechanism to achieve something like this. You can build it on the application side in the following way:
1. Include source offset as one of the columns in the aggregating table.
2. Subscribe to both source and aggregating streams.
3. Block processing of a new row at offset N until the subscriber observes that offset in the aggregating stream.

**Comment by iljaiwjew** [comment]
@Gerrrr thank you for explanation!

**Comment by Gerrrr** [comment]
Glad I could help. I am going to close the issue for now. Feel free to re-open it if you have more questions.
